polygeist/lib/Dialect/Polygeist/Transforms/SYCLRaiseHost.cpp

```diff
@@ -291,6 +291,202 @@ struct AccessorInvokeConstructorPattern
       AccessorTypeTag>::RaiseInvokeConstructorBasePattern;
 };
 
+template <typename TypeTag>
+class RaiseArrayConstructorBasePattern
+    : public OpRewritePattern<LLVM::StoreOp> {
+public:
+  using OpRewritePattern<LLVM::StoreOp>::OpRewritePattern;
+
+  LogicalResult matchAndRewrite(LLVM::StoreOp op,
+                                PatternRewriter &rewriter) const final {
+    // The 'this*' is the address to which the element stores are performed.
+    auto alloc = dyn_cast_or_null<LLVM::AllocaOp>(op.getAddr().getDefiningOp());
+
+    if (!alloc)
+      return failure();
+
+    assert(alloc.getElemType().has_value() &&
+           "Expecting element type attribute for opaque alloca");
```

Comment:
If the device side is using typed pointers, we might hit this assert. Return `failure()` insted.